docs/architecture/adr-019-subspace-post-moving.md

```diff
@@ -0,0 +1,77 @@
+# ADR 019: Subspace post migration
+
+## Changelog
+- April 20th, 2023: First draft;
+
+## Status
+
+Proposed
+
+## Abstract
+
+This ADR introduces a new feature that enables post owners to move their own content between subspaces.
+
+## Context
+
+Currently, the only way to move a post to another subspace is by following process:
+1. Create an identical post as the original post in the target subspace
+2. Delete the original post on source subspace
+
+This process is overly complicated and time-consuming because it requires the user to perform two steps and manually recreate the post in the new location. Additionally, it can lead to inconsistencies if the user forgets to include certain details. A more user-friendly process would be beneficial for improving efficiency and reducing the risk of inconsistencies.
+
+## Decision
+
+We will implement a new handler that allows users to move their own posts through a message. Here are the upcoming changes:
```

Comment:
There is one thing we need to consider: implementing this new message, when moving a post to another subspace, should we 
1. migrate all the reports, reactions, and everything else that references that post as well? OR
2. should we just delete everything in the original subspace instead?

Reply:
We should deleted everything, except for reference posts inside the original subspace since:
1. Supported reactions is not compatible between subspaces, for instance, `A` subspace registered reaction id 1 can be different from `B` subspace one.
2. Supported report reasons is not compatible between subspaces, for instance, `A` subspace supported reason id 1 can be different from `B` reason id

Also, tracking a post reactions and reports history can be done with finding `MsgMovePost` event to get the original post id, then check all the actions related to the post.

Therefore, I think it would be no problem to delete all the references except for posts.

Reply:
Updated to ADR

Reply:
Should we move comments and everything else, when moving a post? Because if yes, then it would mean that the owner of a post owns also all of its comments, which might not be want we want. 

If I comment a post inside Scripta, and then the owner moves it to anohter subspace, I don't want that person to control my comment and move it to the subspace as well. It's my content and I should always be in control. If I created it inside Scripta, I would prefer it being deleted instead of migrated without my consent. 

I think we should only migrate the post and delete all the references to it.

Reply:
Agree with you, we should remain the references (reactions, reports, and etc.), except for reference posts (reply, reference). The moved target will be only the post controlled by the mover.

Reply:
@dadamu Actually, I think we should *delete* everything from the origin subspace. 

Suppose we are moving a post from `Subspace A` to `Subspace B`. Then we should:

1. delete all reactions and reports that reference that post, within `Subspace A`
2. reset the post `PostReferences` field to be empty 
3. reset the post `ConversationID` to be `0` 
4. set the proper `PostID` within `Subspace B`

In short, after moving the post to `Subspace B` it should appears as if it was a post that was just created. Nothing referencing it should stay inside `Subspace A`, and the post should not have any other posts' references inside it.

Reply:
@RiccardoM My bad, I made a typo that should be __remove__ instead of __remain__, I wanted to say that we should remove all the references, and only remains the comments and posts which refer to the moved post, then delete all the incompatible stuff. 😅

Totally agree with you, I will update ADR tomorrow.

Reply:
@RiccardoM Updated